### ISSUE TITLE
Swift Highlighting Broken

### ISSUE THREAD

**Issue by SimplyDanny** [github-linguist/linguist#6643]
## Describe the bug

With the v7.28.0 release, the syntax highlighting for Swift code makes some files unreadable. See [this file](https://github.com/realm/SwiftLint/blob/main/Source/SwiftLintBuiltInRules/Rules/Lint/UnusedSetterValueRule.swift) in the [SwiftLint](https://github.com/realm/SwiftLint) repository as an example.

Many parts get a red background color in dark mode, which makes it very hard to work with them, especially when the confusing highlighting takes action in code reviews where only a small (red) part is visible. My current solution is to review the changes locally which is uncomfortable.

If a quick fix cannot be found, I'd vote for a rollback to the previous Swift grammar.

FYI: @jtbandes

**Comment by jtbandes** [comment]
Looks like with this file you are encountering a combination of https://github.com/jtbandes/swift-tmlanguage/issues/13 & https://github.com/jtbandes/swift-tmlanguage/issues/7

**Comment by SimplyDanny** [comment]
> Looks like with this file you are encountering a combination of https://github.com/jtbandes/swift-tmlanguage/issues/13 & https://github.com/jtbandes/swift-tmlanguage/issues/7

Yes, it seems like. This is especially a serious problem for SwiftLint where almost all files contain string blocks in one or the other way.

**Comment by jtbandes** [comment]
I'll try to resolve these soon but unfortunately I don't have a way of testing GitHub's highlighting aside from waiting for the next release (https://github.com/github-linguist/linguist/pull/6603#issuecomment-1854708886).

**Comment by lildude** [comment]
Closing as syntax highlighting issues should be opened on the upstream repos as per the [troubleshooting doc](https://github.com/github-linguist/linguist/blob/master/docs/troubleshooting.md#theres-a-problem-with-the-syntax-highlighting-of-a-file).

**Comment by SimplyDanny** [comment]
> Closing as syntax highlighting issues should be opened on the upstream repos as per the [troubleshooting doc](https://github.com/github-linguist/linguist/blob/master/docs/troubleshooting.md#theres-a-problem-with-the-syntax-highlighting-of-a-file).

Fair enough. However, I opened the issue here because I propose to revert the change until the problems are fixed. It's really annoying to work on code reviews at the moment.

**Comment by jtbandes** [comment]
As a temporary suggestion, if you are able to use the VS Code's [GitHub integration](https://code.visualstudio.com/docs/sourcecontrol/github) to do code reviews, you might have a better experience as VS Code is able to use the same grammar without experiencing the errors.

**Comment by LePips** [comment]
+1 to reverting until problems are fixed, it's been really hard to read code for the past bit

**Comment by ZevEisenberg** [comment]
Another +1 to fixing or reverting this soon. It makes diffs like [this](https://github.com/pointfreeco/swift-dependencies/pull/167/files) pretty hard to understand:

<img width="1064" alt="image" src="https://github.com/github-linguist/linguist/assets/464574/06e44285-721b-4a2e-8fa7-b43cc7a8d2f2">

**Comment by jtbandes** [comment]
The latest discussion on the potential issue with GitHub's highlighter has been happening on https://github.com/github-linguist/linguist/pull/6603, but @lildude I wonder if we should reopen this issue (or create a new one) to have a more discoverable place for discussion than on a merged PR?

**Comment by SimplyDanny** [comment]
Guys, really ... it has been over a month now that the highlighting is broken. I appreciate all your work and I very well know that a fix usually is not ready within a few minutes/hours/days. However, a simple revert to the previous state of the highlighter would have been perfectly sufficient to allow people to work with Swift code on GitHub again. Honestly, (re)viewing Swift code at the moment is simply impossible and has been for over a month.

I ask you again to please roll back the grammar change! 🙏

**Comment by lildude** [comment]
> I kindly ask you again to roll back the grammar change! 🙏

This is out of my control as is not a problem with the grammar... it's a problem with the highlighting engine which coincided with the last release. Rolling back the release will make no difference. The team responsible for the highlighting engine is aware of the issue and looking into it. 

Feel free to open a support ticket referencing https://github.com/github-linguist/linguist/issues/6668 if you want to add more to the known issue for passing on to the relevant team.
